lib/Sema/TypeCheckDeclObjC.cpp

```diff
@@ -1886,8 +1890,7 @@ bool swift::fixDeclarationObjCName(InFlightDiagnostic &diag, ValueDecl *decl,
     return false;
   }
 
-  // Determine the Objective-C name of the declaration.
-  ObjCSelector name = *decl->getObjCRuntimeName();
+  auto name = *nameOpt;
   auto targetName = *targetNameOpt;
```

Comment:
I was a little confused by these two lines when I read them. I guess the idea is that these should only be `None` when the declaration is implicit or a subscript? If so, maybe there should be a comment here or a sentence in the doc comment to that effect.